### PR TITLE
Improve nonbonded decoupling protocol for single topology transformations

### PR DESCRIPTION
**Pull request by mcwitt**
The schedule used to determine the 4-d coordinate offset for nonbonded decoupling in single topology calculations is currently suboptimal in a way that is easy to fix.

Brief background:
* Before #893, decoupling of dummy atoms was controlled by a single alchemical schedule `w(lam)`. The 4-d coordinates for dummy atoms were then determined by `w_a(lam) = w(lam)` for dummy atoms in A, or `w_b(lam) = cutoff - w(lam)` for dummy atoms in B. The latter functional form was hardcoded.
* Hence, the single alchemical schedule was constrained to be symmetric (e.g. needed to "slow" near $\lambda=0$, where dummy atoms in A decouple, and also "slow" near $\lambda=1$, where dummy atoms in B couple).
* Since #893, we no longer have this functional form imposed on us. In #900 we opted to just maintain the status quo, using the existing symmetric pre-optimized schedule and reproducing the same functional form for `w_a` and `w_b` (even though there was no requirement for `w_a` and `w_b` to each be symmetric).

This PR modifies the existing 4-d decoupling schedule to be more gradual, taking advantage of the fact that `w_a` and `w_b` no longer need to be individually symmetric. The modification was done by taking the relevant half of the existing pre-optimized schedule depending on whether the atom in question is being coupled or decoupled.

decoupling_protocol

### CODE REVIEW

File: timemachine/fe/lambda_schedule.py [hunk 1]
```diff
@@ -74,9 +75,34 @@ def construct_pre_optimized_relative_lambda_schedule(n_windows):
     np.array of size n_windows
     """
     lambda_schedule = np.array(
-        [0.0, 0.02, 0.04, 0.06, 0.07, 0.08, 0.11, 0.13, 0.15, 0.17, 0.18, 0.20, 0.25, 0.32, 0.42]
+        [0.0, 0.04, 0.08, 0.12, 0.14, 0.16, 0.22, 0.26, 0.3, 0.34, 0.36, 0.4, 0.5, 0.64, 0.84, 1.0]
```
Comment (maxentile) [inline timemachine/fe/lambda_schedule.py:78]: provenance: Is this the previous lambda_schedule * 2 (with 1.0 appended), or was this the result of further optimization?

**Review by maxentile — APPROVED** [review]

File: timemachine/fe/lambda_schedule.py (hunk 1 above)
Comment (mcwitt) [reply timemachine/fe/lambda_schedule.py:78]: Right, this is the previous lambda schedule * 2 with 1.0 appended.

File: timemachine/fe/lambda_schedule.py [hunk 2]
```diff
@@ -74,9 +75,34 @@ def construct_pre_optimized_relative_lambda_schedule(n_windows):
...
+def symmetrize_lambda_schedule(lambda_schedule: NDArray) -> NDArray:
+    ls = lambda_schedule
+    ls = ls[:-1]
+    ls = 0.5 * ls
+    ls = np.concatenate([ls, 1.0 - ls[::-1]])
+    return ls
+
+
+def construct_symmetric_pre_optimized_relative_lambda_schedule(n_windows: int | None):
+    """[DEPRECATED] Symmetrized version of :py:func:`construct_pre_optimized_relative_lambda_schedule`
```
[the remark this answers is not in the document]
Comment (mcwitt) [reply timemachine/fe/lambda_schedule.py]: There had been one usage in a test, but it looks like that was removed in https://github.com/proteneer/timemachine/commit/433e071c46f9544ba9887fe6c854ab321b5b5d48. Removed in https://github.com/proteneer/timemachine/pull/1340/commits/336e87f9521778b49295a0ff2e034264ad3dca32

File: timemachine/fe/lambda_schedule.py [hunk 3]
```diff
@@ -57,7 +58,7 @@ def construct_pre_optimized_absolute_lambda_schedule_solvent(num_windows, nonbon
     return lambda_schedule
 
 
-def construct_pre_optimized_relative_lambda_schedule(n_windows):
+def construct_pre_optimized_relative_lambda_schedule(n_windows: int | None):
```
[the remark this answers is not in the document]
Comment (mcwitt) [reply timemachine/fe/lambda_schedule.py]: No difference (think I was just being lazy and avoiding the import). Changed for consistency in https://github.com/proteneer/timemachine/pull/1340/commits/882cacc0dd2a3b266fe572f3fcc8a73c54f166af

**Review by jkausrelay — APPROVED** [review]

**Pull request merged**